.github/workflows/Intel.yml

```diff
@@ -0,0 +1,71 @@
+name: Intel
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  Intel:
+    strategy:
+      matrix:
+        backend: ['SERIAL']
+        cmake_build_type: ['Release']
+        kokkos_ver: ['develop']
+    runs-on: ubuntu-20.04
+    container: fedora:latest
+    steps:
+      - name: Checkout code    
+        uses: actions/checkout@v2.2.0
+      - name: Install Deps
+        run: |
+          sudo dnf -y update
+          sudo dnf -y install make cmake hwloc-devel
+      - name: Install Intel packages
+        run: |
+          curl -O http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/16756/parallel_studio_xe_2020_update2_professional_edition.tgz
+          tar -xf parallel_studio_xe_*.tgz
+          pushd parallel_studio_xe_*/rpm
+          printf "[icc]\nname=icc\nbaseurl=$PWD\nenabled=1" | sudo tee -a /etc/yum.repos.d/icc.repo
+          popd
+          sudo dnf -y update
+          sudo dnf --nogpgcheck -y install intel-parallel-studio-xe-icc intel-parallel-studio-xe-mkl intel-parallel-studio-xe-ifort
+          echo "LD_LIBRARY_PATH=/opt/intel/lib/intel64:/opt/intel/mkl/lib/intel64" >> $GITHUB_ENV
+          echo "/opt/intel/bin" >> $GITHUB_PATH
```

Comment:
@dalg24 this part, downloading/installing icc, takes 30 min, so if we can create a base container and store it on Github's ghcr.io that would help a lot. We could use the same container to test kokkos with icc as well.